yt_dlp/extractor/spreaker.py

```diff
@@ -94,28 +95,33 @@ class SpreakerIE(InfoExtractor):
     }]
 
     def _real_extract(self, url):
-        episode_id = self._match_id(url)
+        episode_id, key = self._match_valid_url(url).group(
+            'id', 'key')
+        query = {}
+        if key is not None:
+            query = {'key': key}
         data = self._download_json(
             f'https://api.spreaker.com/v2/episodes/{episode_id}',
-            episode_id)['response']['episode']
+            episode_id,
+            query=query)['response']['episode']
         return _extract_episode(data, episode_id)
 
 
 class SpreakerPageIE(InfoExtractor):
-    _VALID_URL = r'https?://(?:www\.)?spreaker\.com/user/[^/]+/(?P<id>[^/?#&]+)'
+    _VALID_URL = r'https?://(?:www\.)?spreaker\.com/episode/(?P<title>[^/?#]+)-(?P<id>\d+)(?:\?key=(?P<key>[^&]+))?'
```

Comment:
I don't get why we need a separate extractor for the pages. Can't we just match the page URL in `SpreakerIE`?
```py
class SpreakerIE(InfoExtractor):
    _VALID_URL = [
        r'''(?x)
        https?://
            api\.spreaker\.com/
            (?:
                (?:download/)?episode|
                v2/episodes
            )/
            (?P<id>\d+)
        ''',
        r'https?://(?:www\.)?spreaker\.com/episode/[^#?/]*(?P<id>\d+)/?(?:[?#]|$)',
    ]
```
If you do that, please also add a full test and not just a only_matching one.